Amended Virtual Event API Docs (#21541)
microsoftgraph/microsoft-graph-docs: concepts/cloud-communication-online-meeting-application-access-policy.md

---
title: "Configure an application access policy using the cloud communications API"
description: "Use the cloud communications API in Microsoft Graph to configure an access policy that allows applications to access cloud communications resources."
author: "awang119"
ms.localizationpriority: medium
ms.prod: "cloud-communications"
---

# Configure application access to online meetings or virtual events

> Besides Graph application permissions (see below [section](#supported-permissions-and-additional-resources) for a list of supported application permissions), the tenant administrators needs to configure an application access policy in order for apps to access cloud communications resources with application permissions. Below are the two main scenarios that require an application access policy. Check the "Permissions" section on each API page to find out if the API requires this policy.

Reviewer: ```suggestion
In addition to the Microsoft Graph application permissions, tenant administrators need to configure an application access policy for apps to access cloud communications resources with application permissions.
```

---
title: "Configure an application access policy using the cloud communications API"
description: "Use the cloud communications API in Microsoft Graph to configure an access policy that allows applications to access cloud communications resources."
author: "awang119"
ms.localizationpriority: medium
ms.prod: "cloud-communications"
---

# Configure application access to online meetings or virtual events

> In addition to the Microsoft Graph application permissions, tenant administrators need to configure an application access policy for apps to access cloud communications resources with application permissions

Reviewer 2: Does this set the appropriate amount of context for the user? You deleted the introductory statement. This first sentence doesn't seem to set the context. For example, why do they need to configure an application access policy? Also note that we're missing a period at the end of the sentence.

Author: I adjusted the order of this sentence. Hope it would make things clearer. Added the missing period

Author: Basically what we are trying to say is, normally you can call Graph APIs with application permission if they support it, but for some of our API, application permission itself solely is not enough, you need application permission+application access policy

For more information, see [supported application permissions](#supported-permissions-and-additional-resources).

The following are the two main scenarios that require an application access policy. 

> **1. Allow applications to access online meetings on behalf of a user:**

Reviewer 2: These don't need to be numbered because they're not sequential steps. I wonder if each should be styled as a heading instead (without a colon); otherwise, I recommend a bulleted list.

Author: changed to h2 heading

> In some cases, such as for background services or daemon apps that run on a server without the presence of a signed-in user, it is appropriate for an app to call Microsoft Graph to take actions on behalf of a user. For example, an app might need to call Microsoft Graph to schedule multiple meetings based on published schedules (such as courses) or external scheduling tools. In these cases, the user that the application acts on behalf of can be anyone so make sure that user has proper privilege (e.g. is the organizer/co-organizer) to access the online meeting.

Reviewer: ```suggestion
In some cases, such as background services or daemon apps running on a server without a signed-in user, it is acceptable for an app to call Microsoft Graph and perform actions on behalf of a user.
For example, an app may require Microsoft Graph to schedule multiple meetings based on published schedules (like courses) or external scheduling tools.
In such situations, the application can act on behalf of any user. Therefore, it is important to make sure that the user possesses the necessary privileges, such as being the organizer or co-organizer, to access the online meeting.
```

**2. Allow applications to access virtual events created by the user:**

> In cases where a signed-in user is not presented, an app can call Microsft Graph to access a virtual event using application permissions. For example, an app can call Microsoft Graph to look up a virtual event that has been created by a user, or retrieve attendance reports of a virtual event that has been created by the user,  without using that user's delegated permissions. In these case, the user must be the organizer of that virtual event.

Reviewer: ```suggestion
In cases where a signed-in user is not presented, an app can call Microsoft Graph to access a virtual event using application permissions. For example, an app can call Microsoft Graph to look up a virtual event that a user has created or retrieve attendance reports of a virtual event created by the user without using that user's delegated permissions. In these cases, the user must be the organizer of that virtual event.
```

Follow these steps to configure an application access policy for cloud communications resources such as online meetings and virtual events. These steps do not apply to other Microsoft Graph resources.

> ## Compare

Reviewer 2: We need a more clear heading here - this is a verb but what is the subject and object? Who is comparing and what are they comparing?

Author: updated

> This article covers the basic steps to configure an application access policy. These steps are specific to cloud communications resources (online meetings, virtual events) and do not apply to other Microsoft Graph resources.

Reviewer: ```suggestion
Follow these steps to configure an application access policy for cloud communications resources such as online meetings and virtual events. These steps do not apply to other Microsoft Graph resources.
```

> In the following table, there are three application access policies:

Reviewer: ```suggestion
The following table specifies three application access policies:
```

> The following table is an example of three application access policies:

Reviewer 2: We need to clarify the text here. Maybe something like: "The following table compares online meeting and virtual event access policies for various scenarios involving two users. These scenarios involve the following application access policies:

- Policy 1 contains...
- Policy 2 contains...
- Policy 3 contains...

   ```

> [!NOTE]
>
> - _Identity_ refers to the policy name when creating the policy, but the user ID when granting the policy.

> > - Granting the policy will take effect for both Online meeting and Virtual event. It's recommended to have two applications if you wish to manage Online meeting and Virtual event separately.

Reviewer: ```suggestion
> - When granting the policy, it will be applied to both Online meetings and Virtual events. If you prefer to manage Online meetings and Virtual events separately, it is advisable to have two separate applications.
```

> > - When granting the policy, it will be applied to both Online meetings and Virtual events. If you prefer to manage Online meetings and Virtual events separately, it is advisable to have two separate applications.

Reviewer: ```suggestion
> - When granting the policy, it will be applied to both online meetings and virtual events. If you prefer to manage online meetings and virtual events separately, it is advisable to have two separate applications.
```

> > - When granting the policy, it will be applied to both online meetings and virtual events. If you prefer to manage online meetings and virtual events separately, it is advisable to have two separate applications.

Reviewer 2: Change "it is advisable to have" to "we recommend that you have".

- OnlineMeetings.ReadWrite.All
- OnlineMeetingArtifact.Read.All

> and/or to control virtual event access for an app that has been granted the following application permissions:

Reviewer 2: I'm not sure why we're separating this permission from the rest of the list, but starting the sentence with and/or is not Microsoft style and will be confusing to a global audience. Can we just have one list of all the permissions? You might do something like this:

VirtualEvent.Read.All (for access to virtual events)

Author: Done

- VirtualEvent.Read.All

For more information about configuring application access policy, see the [PowerShell cmdlet reference for New-ApplicationAccessPolicy](/powershell/module/skype/new-csapplicationaccesspolicy).

## Errors

> If you attempt an API call to access an online meeting or virtual event without configuring the application access policy, you may encounter the following error: 

Reviewer 2: Please change "may" to "might" (may implies permission).

```json
{
    "error": {
        "code": "Forbidden",

>         "message": "No Application Access Policy found for this app",

Reviewer: ```suggestion
        "message": "No application access policy found for this app",
```

>         "message": "No Application Access Policy found for this app",

Reviewer: @frankpeng7, is this exactly how the error will be displayed, or can I update the text to meet our guidelines? We use sentence casing.
